### PR TITLE
Chapter14

### PR DESCRIPTION
**Pull request by NurAnisAqilah**

### CODE REVIEW

**Comment by NurAnisAqilah** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -76,7 +76,10 @@ def password_reset_expired?
   end
 
   def feed
-    microposts
+    following_ids = "SELECT followed_id FROM relationships
+                    WHERE follower_id = :user_id"
+    Micropost.where("user_id IN (#{following_ids})
```
Comment (ngocvt-0484) [inline app/models/user.rb:81]: please use scope
in micropost.rb
```
   scope :by_user, ->(ids){where user_id: ids}
```
and call it here
Micropost.by_user following_ids << id

**Comment by NurAnisAqilah** [comment]
ready

File: app/models/user.rb (hunk 1 above)
Comment (manhnt-0960) [reply app/models/user.rb:81]: please fix it before ready

**Comment by NurAnisAqilah** [comment]
ready

File: app/controllers/relationships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  def create
+    @user = User.find_by id: params[:followed_id]
+    if @user.present?
+      current_user.follow @user
+    else
+      @error = t "follow.fail"
+    end
+    respond_to :js
+      format.js
+    end
```
Comment (saobv-2223) [inline app/controllers/relationships_controller.rb:13]:
```suggestion
    respond_to :js
```

File: app/controllers/relationships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    relationship = Relationship.find_by id: params[:id]
+    if relationship.present?
+      @user = relationship.followed
+      current_user.unfollow @user
+    else
+      @error = t "follow.fail"
+    end
+    respond_to do :js
+      format.js
+    end
```
Comment (saobv-2223) [inline app/controllers/relationships_controller.rb]:
```suggestion
    respond_to :js
```

**Comment by NurAnisAqilah** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
Please, fix comment firstly.

**Comment by NurAnisAqilah** [comment]
ready

File: config/routes.rb [hunk 4]
```diff
@@ -1,23 +1,20 @@
...
-  resources :account_activations, only: :edit
-  resources :password_resets, except: %i(index show destroy)
-  resources :microposts, only: [:create, :destroy]
-  resources :microposts, except: [:index, :new, :edit, :show, :update]
-
-  get "/login", to: "sessions#new"
-  post "/login", to: "sessions#create"
-  delete "/logout", to: "sessions#destroy"
-  
+    root "static_pages#home"
```
Comment (manhnt-0960) [inline config/routes.rb]: no need indent here

File: app/views/users/_follow_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<% unless current_user?(@user) %>
+  <div id="follow_form">
+  <% if current_user.following?(@user) %>
```
Comment (manhnt-0960) [inline app/views/users/_follow_form.html.erb]: wrong indent

File: app/models/user.rb (hunk 1 above)
Comment (manhnt-0960) [reply app/models/user.rb:81]: call it here
Micropost.by_user following_ids << id

**Comment by NurAnisAqilah** [comment]
ready

**Comment by ngocvt-0484** [comment]
not fix comment yet

**Pull request merged**